woocommerce/woocommerce
# [Settings] Implement URL routing with query parameters and active sidebar item

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR enhances sidebar by handling URL query parameters to update the sidebar accurately, ensuring the active item reflects the current route. The routing logic now dynamically updates based on query parameters, improving navigation consistency and usability.

Besides, this PR tweaks the variable names to be more consistent and descriptive and remove the unnecessary `{ color: 'white' }` style from `NotFound` component.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Visit site and turn on the feature flag `settings`, `Tools > WCA Test Helper > Features`.
2. Install/Activate Gutenberg.
3. Visit the new Settings, `WooCommerce > Settings`.
4. See the new layout and sidebar populated with items.
5. Click on different sidebar items and see the content changes accordingly.
6. Confirm the URL query parameters are updated correctly. (e.g. `?page=wc-settings&tab=general`, `?page=wc-settings&tab=products`, etc.)


https://github.com/user-attachments/assets/f46713ca-7abb-4e62-9cf6-bc9659ed4c86




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
